src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerContext.cs

```diff
@@ -94,7 +86,9 @@ protected JsonSerializerContext(JsonSerializerOptions? options)
             if (options != null)
             {
                 options.VerifyMutable();
-                Options = options;
+                options.TypeInfoResolver = this;
+                options.MakeReadOnly();
+                _options = options;
```

Comment:
The comment above about AddContext might need some re-wording

Reply:
Let's revisit when we update the API docs.